### PR TITLE
Fix issue #11

### PR DESCRIPTION
**Pull request by mohanish3**
AboutFestActivity created with data retrieval from Firebase

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -23,19 +25,20 @@
...
+            android:label="@string/title_activity_eateries"
             android:parentActivityName=".home.HomeActivity">
             <meta-data
                 android:name="android.support.PARENT_ACTIVITY"
                 android:value="com.macbitsgoa.events.home.HomeActivity" />
         </activity>
+        <activity android:name=".aboutfest.AboutFestActivity"></activity>
     </application>
 
-</manifest>
```
Comment (Rushi98) [inline app/src/main/AndroidManifest.xml]: newline here

File: app/src/main/java/com/macbitsgoa/events/aboutfest/AboutFestActivity.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+public class AboutFestActivity extends FragmentActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_about_fest);
+
+        //Initialize Fresco
+        Fresco.initialize(this);
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/aboutfest/AboutFestActivity.java]: put this initialization in application class 
> app/src/main/java/com/macbitsgoa/events/Events.java
```
public class Events extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        Fresco.initialize(this);
    }
}
```
Also add the
```xml
      android:name=".Events"
```
attribute in `AndroidManifest.xml` file.

File: app/src/main/java/com/macbitsgoa/events/aboutfest/AboutFestActivity.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+        ViewPager viewPager = (ViewPager)findViewById(R.id.about_fest_viewpager);
+        viewPager.setAdapter(new CustomPagerAdapter(getSupportFragmentManager()));
+
+        TabLayout tabLayout = (TabLayout)findViewById(R.id.tab_layout);
+        tabLayout.setupWithViewPager(viewPager, true);
+    }
+
+    private class CustomPagerAdapter extends FragmentPagerAdapter {
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/aboutfest/AboutFestActivity.java]: put add "inner" classes in separate files

File: app/src/main/java/com/macbitsgoa/events/aboutfest/DescriptionFragment.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        databaseReference.child("description").addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                String value = dataSnapshot.getValue(String.class);
+                description.setText(value);
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+                Log.d("Tag", "Error");
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/aboutfest/DescriptionFragment.java]: log this error as
```java
// at the start of class
private static final String TAG = DescriptionFragment.class.getSimpleName();

// in onCancelled method
Log.e(TAG, databaseError.getMessage(), databaseError.toException()
```
(I don't remember exact methods, but you get the idea)

File: app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersAdapter.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+
+import com.facebook.drawee.view.SimpleDraweeView;
+import com.macbitsgoa.events.R;
+
+import java.util.ArrayList;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class OrganisersAdapter extends RecyclerView.Adapter<OrganisersAdapter.ViewHolder> {
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersAdapter.java]: break the line before `extends`. (Else we need to scroll horizontally to see full line)

File: app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersFragment.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.recyclerview.widget.RecyclerView;
+
+public class OrganisersFragment extends Fragment {
+    private RecyclerView.Adapter adapter;
+    private ArrayList<OrganisersList> organisersList = new ArrayList<>();
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view;
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersFragment.java]: make all variables private by default

File: app/src/main/java/com/macbitsgoa/events/home/HomeActivity.java [hunk 7]
```diff
@@ -47,6 +48,15 @@ private void populateFeatures() {
                     .commit();
         }
 
+        featuresFragManager
```
Comment (Rushi98) [inline app/src/main/java/com/macbitsgoa/events/home/HomeActivity.java]: use `BuildConfig.aboutFest` as done for eateries

File: app/src/main/res/layout/fragment_about_fest_organisers.xml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+        <!--androidx.cardview.widget.CardView
```
Comment (Rushi98) [inline app/src/main/res/layout/fragment_about_fest_organisers.xml]: delete these "comments"

File: app/build.gradle [hunk 9]
```diff
@@ -42,5 +42,12 @@ dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
     implementation "androidx.constraintlayout:constraintlayout:1.1.0"
     implementation "com.google.android.material:material:1.0.0-alpha1"
+    implementation 'com.google.firebase:firebase-database:15.0.1'
```
[the remark this answers is not in the document]
Comment (mohanish3) [reply app/build.gradle]: Android studio says that the latest db version is 15.0.1

File: app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersFragment.java (hunk 6 above)
Comment (mohanish3) [reply app/src/main/java/com/macbitsgoa/events/aboutfest/OrganisersFragment.java]: Conflicts with codacy to keep the variable local in this case

File: app/build.gradle [hunk 10]
```diff
@@ -42,5 +44,11 @@ dependencies {
...
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0-alpha1'
+    implementation 'androidx.legacy:legacy-support-v13:1.0.0-alpha1'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
```
Comment (Rushi98) [inline app/build.gradle]: delete this line as constraint layout is already included

**Pull request merged**